app/src/lib/stores/helpers/infer-compare-to-branch.ts

```diff
@@ -0,0 +1,153 @@
+import { Branch, IAheadBehind } from '../../../models/branch'
+import { PullRequest } from '../../../models/pull-request'
+import { getAheadBehind, revRange } from '../../git'
+import { Repository } from '../../../models/repository'
+import { GitHubRepository } from '../../../models/github-repository'
+
+interface IOfGithub {
+  kind: 'github'
+  repository: Repository
+}
+
+interface IOfPullRequest {
+  kind: 'pr'
+  pullRequest: PullRequest
+}
+
+interface IOfFork {
+  kind: 'fork'
+  repository: Repository
+  gitHubRepository: GitHubRepository
+  currentBranch: Branch
+}
+
+// I am struggling to come up with a decent name for this type
+type NamingIsHard = IOfPullRequest | IOfGithub | IOfFork
+
+/**
+ * Infers which branch to use as the comparison branch
+ *
+ * @param branches The list of all branches for the repository
+ * @param needsAName The object that determines how the branch will be inferred
+ */
+export async function inferComparisonBranch(
+  branches: ReadonlyArray<Branch>,
+  needsAName: NamingIsHard
+): Promise<Branch | null> {
+  switch (needsAName.kind) {
+    case 'pr':
+      return _getFeatureBranchOfPullRequest(branches, needsAName.pullRequest)
+    case 'github':
+      return _getDeafultBranchOfGithubRepo(
+        branches,
+        needsAName.repository.gitHubRepository!
+      )
+    case 'fork':
+      return _getDefaultBranchOfFork(
+        branches,
+        needsAName.repository,
+        needsAName.gitHubRepository,
+        needsAName.currentBranch
+      )
+    default:
+      return _getMasterBranch(branches)
+  }
+}
+
+/**
+ * Do **not** call this method; use `inferComparisonBranch` instead
+ *
+ * Returns `master`, the default branch of a Git repository
+ *
+ * @param branches The list of all branches for the repository
+ */
+export function _getMasterBranch(
```

Comment:
These functions are prefixed with an `_` (underscore) because I intend for them to not be used outside of this module; however, I still want the ability to test them. @niik and I spoke on this and he mentioned we could put them in a separate file, but we agreed that we can cross that bridge at a later point in time.